config.go

```diff
@@ -32,6 +40,7 @@ const (
 type FileConfig struct {
 	SkipPageIndex    bool
 	SkipBloomFilters bool
+	MetadataReaders  MetadataReaders
```

Comment:
Rather than having a separate configuration option, would it be meaningful to have the `io.ReaderAt` passed to `OpenFile` implement the other methods?

Something that could be used this way maybe?
```go
if header, ok := r.(interface{ Header() io.ReaderAt }); ok {
    ...
}
```

Reply:
Good idea! I'm just wondering if it would be intuitive for developers using the library to add additional readers this way, I guess its slightly less explicit than adding a defined configuration option. 

However it does make it quite easy to add cached readers. Would you prefer if I restructured it that way?